src/middlewares/user.ts

```diff
@@ -168,9 +168,13 @@ export const checkUserTwoFactorAuthMiddleware = async (
           return res.redirect("/users/2fa-sign-in");
         }
       }
-
       return next();
     } catch (error) {
+      if (error instanceof UserNotFoundError) {
```

Comment:
This code should be moved not duplicated : 

https://github.com/numerique-gouv/moncomptepro/pull/884/files#diff-707a46f9576d844e40f2f3b845cc04e1256deba287ddb3ae120cec44171cadb4R219-R223